drivers/lora/lora_basics_modem/lbm_common.c

```diff
@@ -0,0 +1,549 @@
+	status = ral_set_tx(&config->ralf.ral);
+	if (status != RAL_STATUS_OK) {
+		ret = -EINVAL;
+		goto release;
+	}
+	return 0;
+
+release:
+	modem_release(dev);
+	return ret;
+}
+
+int lbm_lora_send(const struct device *dev, uint8_t *msg, uint32_t msg_len)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	struct k_poll_signal done = K_POLL_SIGNAL_INITIALIZER(done);
+	struct k_poll_event evt =
+		K_POLL_EVENT_INITIALIZER(K_POLL_TYPE_SIGNAL, K_POLL_MODE_NOTIFY_ONLY, &done);
+	uint32_t air_time;
+	int ret;
+
+	/* Trigger the asynchronous send */
+	ret = lbm_lora_send_async(dev, msg, msg_len, &done);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Calculate expected airtime of the packet */
+	air_time = ral_get_lora_time_on_air_in_ms(&config->ralf.ral, &data->pkt_params,
+						  &data->mod_params);
+	LOG_DBG("Expected airtime: %d ms", air_time);
+
+	/* Wait for the packet to finish transmitting.
+	 * Setting up the transaction takes some minimal time, take it into
+	 * account to ensure extremely short packets don't incorrectly timeout.
+	 * Use twice the tx duration to ensure that we are actually detecting
+	 * a failed transmission, and not some minor timing variation between
+	 * modem and driver.
+	 */
+	ret = k_poll(&evt, 1, K_MSEC(10 + (2 * air_time)));
+	if (ret < 0) {
+		if (modem_release(dev)) {
+			LOG_ERR("Packet transmission failed!");
+		} else {
+			/* TX done interrupt is currently running */
+			k_poll(&evt, 1, K_FOREVER);
+		}
+	}
+	return ret;
+}
+
+int lbm_lora_recv(const struct device *dev, uint8_t *msg, uint8_t msg_len, k_timeout_t timeout,
+		  int16_t *rssi, int8_t *snr)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	struct k_poll_signal done = K_POLL_SIGNAL_INITIALIZER(done);
+	struct k_poll_event evt =
+		K_POLL_EVENT_INITIALIZER(K_POLL_TYPE_SIGNAL, K_POLL_MODE_NOTIFY_ONLY, &done);
+	ral_status_t status;
+	int ret;
+
+	/* Ensure available, decremented by op_done_work_handler or on timeout */
+	if (!modem_acquire(dev)) {
+		return -EBUSY;
+	}
+
+	/* Store signal */
+	data->operation_done = &done;
+	data->rx_state.sync.msg = msg;
+	data->rx_state.sync.msg_len = msg_len;
+
+	/* Configure modem for RX */
+	lbm_driver_antenna_configure(dev, MODE_RX);
+	data->modem_mode = MODE_RX;
+
+	/* Start the reception in continuous mode.
+	 * Receive timeouts are handled by the k_poll timeout.
+	 * In theory we should be able to use the one-shot mode here and transition
+	 * back to IDLE slightly faster, but the SX127x driver does not appear to
+	 * receive packets reliably in the single-shot mode.
+	 */
+	status = ral_set_rx(&config->ralf.ral, RAL_RX_TIMEOUT_CONTINUOUS_MODE);
+	if (status != RAL_STATUS_OK) {
+		ret = -EINVAL;
+		goto release;
+	}
+
+	/* Wait for the packet to be received */
+	ret = k_poll(&evt, 1, timeout);
+	if (ret < 0) {
+		if (modem_release(dev)) {
+			LOG_INF("Receive timeout");
```

Comment:
log ret, should this be a LOG_WRN or LOG_ERR?

Reply:
Again, the timeout is the only failing return code, which is adequately described by the message itself.

I don't think this needs to be `WRN` or `ERR`, since there is no guarantee that the caller actually expected to receive a message (it could just be a trailing RX window after TX). If the user considers this an error, they can add their own logging at the application layer.